### PR TITLE
CI - Add code sniffer

### PR DESCRIPTION
**Pull request by timotei-litespeed**

### CODE REVIEW

**Review by tynanbe — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,19 +8,36 @@ on:
...
           cache: npm
           cache-dependency-path: package-lock.json
 
-      - run: npm ci
+      # Start actions
+      - run: mkdir -p .cache
+      - run: composer install
+      - run: composer run sniff-check
 
+      - run: npm ci
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .github/workflows/ci.yml:42]: Should I do something here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,19 +8,36 @@ on:
       - 'v*.*'
   pull_request:
 
+# Cancels all previous workflow runs for pull requests that have not completed.
+# concurrency:
+#   group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+#   cancel-in-progress: true
+
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .github/workflows/ci.yml]: Removed

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -8,19 +8,36 @@ on:
...
 jobs:
-  test:
+  check:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
 
+      # Initialize Cache Composer
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .github/workflows/ci.yml]: Removed

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -8,19 +8,36 @@ on:
...
+      - name: Cache Composer dependencies
+        uses: actions/cache@v3
+        with:
+          path: .cache
+          key: ${{ runner.os }}-${{ hashFiles('**/composer.lock') }}
+
+      # Initialize Node.js
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .github/workflows/ci.yml]: Changed to name

File: .prettierrc [hunk 5]
```diff
@@ -2,14 +2,20 @@
 	"overrides": [
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .prettierrc]: Reverted to previous version

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+	"name": "litespeedtech/lscache_wp",
+	"require-dev": {
+		"squizlabs/php_codesniffer": "*",
+		"phpcompatibility/php-compatibility": "*"
+	},
+	"prefer-stable": true,
+	"scripts": {
+		"post-install-cmd": "phpcs --config-set installed_paths vendor/phpcompatibility/php-compatibility",
+		"post-update-cmd": "phpcs --config-set installed_paths vendor/phpcompatibility/php-compatibility",
+		"sniff-check": "phpcs --standard=./phpcs.xml.dist"
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply composer.json]: Works and changed

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tynanbe) [reply .github/workflows/ci.yml:36]:
```yml
      - run: npm clean-install
```

File: .prettierrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (tynanbe) [reply .prettierrc]: The file shouldn't show up at all in https://github.com/litespeedtech/lscache_wp/pull/744/files

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .github/workflows/ci.yml]: Changed

File: .prettierrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply .prettierrc]: Fixed

**Comment by timotei-litespeed** [comment]
@szepeviktor  these changes have been applied to the future version of the plugin.
Talking to my colleagues we decided to look into new ways to improve error detection, improve the security and help maintain the plugin easier.
If you're still up for it: what suggestions do you have? Please create a branch from https://github.com/litespeedtech/lscache_wp/tree/v7.0 and add changes into a PR.

Thank you!

**Comment by szepeviktor** [comment]
> what suggestions do you have?

https://github.com/litespeedtech/lscache_wp/pull/742#issuecomment-2351381229

Please consider choosing a coding standard: PSR, WordPress, custom, a minimal.
This is my custom ruleset: https://github.com/szepeviktor/phpcs-psr-12-neutron-hybrid-ruleset

I could add EditorConfig (for Prettier) and syntax error check.

As for static analysis: the plugin should pass PHPStan level 0.

**Pull request merged**